easybuild/easyblocks/i/icc.py

```diff
@@ -105,53 +112,81 @@ def sanity_check_step(self):
+                dirmap = {
+                    'PATH': ['mpi/intel64/bin', 'ipp/bin/intel64', '%s/gdb/intel64/bin' % debuggerpath, 'bin/intel64'],
+                    'LD_LIBRARY_PATH': libpaths,
+                    'LIBRARY_PATH': libpaths,
+                    'MANPATH': ['man/common', 'man/en_US', 'debugger/gdb/intel64/share/man'],
+                    'CPATH': ['daal/include', 'tbb/include', 'mkl/include', 'ipp/include'],
+                    'INTEL_PYTHONHOME': ['%s/python/intel64' % debuggerpath],
+                    'DAALROOT': ['daal'],
+                    'TBBROOT': ['tbb'],
+                    'IPPROOT': ['ipp'],
+                    'CLASSPATH': ['daal/lib/daal.jar'],
+                }
+
 
         # in recent Intel compiler distributions, the actual binaries are
         # in deeper directories, and symlinked in top-level directories
         # however, not all binaries are symlinked (e.g. mcpcom is not)
-        if os.path.isdir("%s/composerxe-%s" % (self.installdir, self.version)):
-            prefix = "composerxe-%s" % self.version
+        # more recent versions of the Intel Compiler (2013.sp1 and newer)
+        if os.path.isdir("%s/composer_xe_%s" % (self.installdir, self.version)):
+            prefix = "composer_xe_%s" % self.version
             oldmap = dirmap
             dirmap = {}
             for k, vs in oldmap.items():
                 dirmap[k] = []
-                if k == "LD_LIBRARY_PATH":
-                    prefix = "composerxe-%s/compiler" % self.version
-                else:
-                    prefix = "composerxe-%s" % self.version
+                prefix = "composer_xe_%s" % self.version
                 for v in vs:
                     v2 = "%s/%s" % (prefix, v)
-                    dirmap[k].append(v2)
+                    if os.path.isdir("%s/%s" % (self.installdir, v2)):
+                        dirmap[k].append(v2)
 
-        elif os.path.isdir("%s/compiler" % (self.installdir)):
-            prefix = "compiler"
+        if os.path.isdir("%s/compilers_and_libraries_%s/linux" % (self.installdir, self.version)):
+            prefix = "compilers_and_libraries_%s/linux" % self.version
             oldmap = dirmap
             dirmap = {}
             for k, vs in oldmap.items():
                 dirmap[k] = []
-                prefix = ''
-                if k == "LD_LIBRARY_PATH":
-                    prefix = "compiler/"
                 for v in vs:
-                    v2 = "%s%s" % (prefix, v)
-                    dirmap[k].append(v2)
+                    v2 = "%s/%s" % (prefix, v)
+                    if os.path.exists("%s/%s" % (self.installdir, v2)):
+                        dirmap[k].append(v2)
 
         return dirmap
+
+    def make_module_extra(self):
```

Comment:
this is already done in `IntelBase` now, shouldn't be needed to override it

(see #635)